src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -1,81 +1,328 @@
-              this.invalidPasswordMessage = error.data.description
-            } else {
-              this.signUpErrors[error.status] = true
+    this.submitting = false
+  }
+
+  $onDestroy () {
+    this.oktaSignInWidget.remove()
+  }
+
+  async setUpSignUpWidget () {
+    const donorData = await this.loadDonorDetails()
+    this.$window.currentStep = 1 // Default to step 1
+
+    this.oktaSignInWidget = new OktaSignIn({
+      ...this.sessionService.oktaSignInWidgetDefaultOptions,
+      assets: {
+        baseUrl: `${this.$window.location.origin}/assets/okta-sign-in/`
+      },
+      flow: 'signup',
+      registration: {
+        parseSchema: (schema, onSuccess) => {
+          const step = this.$window.currentStep || 1
+          // Split the form into multiple steps for better user experience.
+          // Retain the values entered by the user in each step.
+          // Pre-populate the form fields with existing user details to save time.
+          const steps = {
+            // Step 1: Name and email
+            1: [
+              {
+                ...schema[0],
+                value: this.$rootScope.firstName ?? donorData.name['given-name'] ?? this.sessionService.session.first_name ?? ''
+              },
+              {
+                ...schema[1],
+                value: this.$rootScope.lastName ?? donorData.name['family-name'] ?? this.sessionService.session.last_name ?? ''
+              },
+              {
+                ...schema[2],
+                value: this.$rootScope.email ?? donorData.email ?? this.sessionService.session.email ?? ''
+              }
+            ],
+            // Step 2: Address plus phone number
+            2: [
+              {
+                ...schema[3],
+                value: this.$rootScope.streetAddress ?? donorData.mailingAddress.streetAddress ?? ''
+              },
+              {
+                ...schema[4],
+                value: this.$rootScope.city ?? donorData.mailingAddress.locality ?? ''
+              },
+              {
+                ...schema[5],
+                value: this.$rootScope.state ?? donorData.mailingAddress.region ?? ''
+              },
+              {
+                ...schema[6],
+                value: this.$rootScope.zipCode ?? donorData.mailingAddress.postalCode ?? ''
+              },
+              {
+                ...schema[7],
+                value: this.$rootScope.countryCode ?? donorData.mailingAddress.country ?? ''
+              },
+              {
+                ...schema[8],
+                value: this.$rootScope.primaryPhone ?? donorData['phone-number'] ?? ''
+              }],
+            // Step 3: Password
+            3: [schema[8]]
+          }
+          onSuccess(steps[step])
+        },
+        preSubmit: (postData, onSuccess) => {
+          const step = this.$window.currentStep
+          const userProfile = postData.userProfile
+
+          if (step === 1) {
+            Object.assign(this.$rootScope, {
```

Comment:
I don't think this will trigger a digest for `$rootScope`. It's fine though unless we have watchers on `$rootScope`.

Reply:
Here, I don't think I should be using $rootscope. I've switched it to $scope as other components aren't relying on this information.